kobotoolbox/kpi
# Add Enterprise plans to kpi front-end, update Usage page

## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [X] Ensure the tests pass
4. [X] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [X] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Add the Enterprise Gold and Enterprise Platinum plans to the Plans page. Re-style the Usage page and some parts of the Billing page.

## Notes

This PR is for reviewing the front-end changes for this scope - [kpi#4783](https://github.com/kobotoolbox/kpi/pull/4783) targets only the back-end changes, and should be merged first.

A few changes were necessary to get the front-end right:
* Buttons on the billing and usage pages are now Sentence case instead of UPPERCASE
* Added an additional breakpoint definition for 600px
* The billing page now allows hiding/showing groups of plans, based on the `plan_type` metadata on the Stripe product.
* The `AriaText` component now lowercases the screen reader text, to prevent screen readers from saying, for example, 'Kay-obo' when reading the string `Kobo`.

## Testing

Some (not exhaustive) testing steps:
1. Check out the updated usage page at `/#/account/usage` - make sure the styling is in line with the [design](https://www.figma.com/file/NonWWcfadc3QfS3q3D8hRB/%E2%9A%99%EF%B8%8F-KPI-%2F-Account-Settings?type=design&node-id=2580-27554&mode=design&t=wG9q54oqUeu2Fbfq-0). (Some things, like the 'Get add-ons' button, are implemented but won't be used until a later scope.)
1. Purchase an add-on from the plans page (`/#/account/plan`). You might need to create a new user to do so, since storage add-ons are only visible on the Community/Legacy plan. Then go to the usage page and confirm that you see your storage add-on reflected on the Storage tab. Mockup:
![image](https://github.com/kobotoolbox/kpi/assets/7819986/ea5a05c2-1e35-4d3f-a1d3-8209495e74ef)
1. Go to the plans page and confirm that the checkout/manage links work correctly.
1. To check out for the Enterprise plans, go to `/#/account/plan?type=enterprise`. Once you've purchased the Enterprise Plan, navigate back to `/#/account/plan` and confirm that you only see the Enterprise plans.
